Demo/Fullscreen/SearchResultMapView/SearchResultMapViewDemoViewController.swift

```diff
@@ -0,0 +1,115 @@
+//
+//  Copyright © 2019 FINN AS. All rights reserved.
+//
+
+import FinniversKit
+import MapKit
+
+public class SearchResultMapViewDemoViewController: DemoViewController<UIView> {
+
+    private lazy var searchResultMapView: SearchResultMapView = {
+        let view = SearchResultMapView(withAutoLayout: true)
+        view.delegate = self
+        return view
+    }()
+
+    // MARK: - Setup
+
+    override public func viewDidLoad() {
+        super.viewDidLoad()
+        setup()
+    }
+
+    override public func viewDidLayoutSubviews() {
+        // Place a nice looking view in front of the mapView to prevent the UI tests from failing.
+        for subview in searchResultMapView.subviews {
+            guard let mapView = subview as? MKMapView else { continue }
+            let colorfulView = UIView(withAutoLayout: true)
+            colorfulView.backgroundColor = .mint
+            mapView.addSubview(colorfulView)
+            colorfulView.fillInSuperview()
+            DispatchQueue.main.asyncAfter(deadline: .now() + 0.5, execute: { [weak colorfulView] in
+                colorfulView?.removeFromSuperview()
+            })
+        }
+    }
+
+    private func setup() {
+        view.backgroundColor = .milk
+        view.addSubview(searchResultMapView)
+
+        let region = MKCoordinateRegion(center: SearchResultMapViewAnnotationFactory.centerLocation, latitudinalMeters: 1000, longitudinalMeters: 1000)
+
+        searchResultMapView.configure(withInitialRegion: region, andShowingUserLocation: true)
+        searchResultMapView.setMapOverlay(SearchResultMapViewAnnotationFactory.tileOverlay)
+
+        SearchResultMapViewAnnotationFactory.create(numberOfAnnotations: 10).forEach { searchResultMapView.addAnnotation($0) }
+
+        if #available(iOS 11.0, *) {
+            NSLayoutConstraint.activate([
+                searchResultMapView.topAnchor.constraint(equalToSystemSpacingBelow: view.safeAreaLayoutGuide.topAnchor, multiplier: 1.0),
+                searchResultMapView.bottomAnchor.constraint(equalToSystemSpacingBelow: view.safeAreaLayoutGuide.bottomAnchor, multiplier: 1.0),
+                searchResultMapView.leadingAnchor.constraint(equalTo: view.leadingAnchor),
+                searchResultMapView.trailingAnchor.constraint(equalTo: view.trailingAnchor)
+                ])
+        }
+    }
+
+    // MARK: - Private
+
+    private func addMapOverlay() {
+        searchResultMapView.setMapOverlay(SearchResultMapViewAnnotationFactory.tileOverlay)
+    }
+
+    private func clearMapOverlay() {
+        searchResultMapView.clearMapOverlay(SearchResultMapViewAnnotationFactory.tileOverlay)
+    }
+
+}
+
+// MARK: - SearchResultMapViewDelegate
+
+extension SearchResultMapViewDemoViewController: SearchResultMapViewDelegate {
+
+    public func searchResultMapView(_ view: SearchResultMapView, didSelect mapSettingsAction: MapSettingsButton.Actions, in button: MapSettingsButton) {
+        switch mapSettingsAction {
```

Comment:
I think it is enough to call this `action`, its type is enough to know what it is in these types of delegate methods.